de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/ft/FaultTreeDiagramToolBehaviorProvider.java

```diff
@@ -118,4 +128,66 @@ public Object getToolTip(GraphicsAlgorithm ga) {
 		
 		return super.getToolTip(ga);
 	}
+	
+	@Override
+	public IPaletteCompartmentEntry[] getPalette() {
+		final String SIGNATURE = "getPaletteCompartments()";
+		boolean info = T.racer().info();
+		if (info) {
+			T.racer().entering(DefaultToolBehaviorProvider.class, SIGNATURE, new Object[0]);
+		}
+		List<IPaletteCompartmentEntry> compartments = new ArrayList<IPaletteCompartmentEntry>();
+
+		PaletteCompartmentEntry compartmentEntry = new PaletteCompartmentEntry("Connections", null);
+		compartments.add(compartmentEntry);
+
+		IFeatureProvider featureProvider = getFeatureProvider();
+		ICreateConnectionFeature[] createConnectionFeatures = featureProvider.getCreateConnectionFeatures();
+		if (createConnectionFeatures.length > 0) {
+
+			for (ICreateConnectionFeature createConnectionFeature : createConnectionFeatures) {
+				ConnectionCreationToolEntry ccTool = new ConnectionCreationToolEntry(
+						createConnectionFeature.getCreateName(), createConnectionFeature.getCreateDescription(),
+						createConnectionFeature.getCreateImageId(), createConnectionFeature.getCreateLargeImageId());
+				ccTool.addCreateConnectionFeature(createConnectionFeature);
+	
+				compartmentEntry.addToolEntry(ccTool);
+			}
+
+		}
+
+		compartmentEntry = new PaletteCompartmentEntry("Gates", null);
+		compartments.add(compartmentEntry);
+
+		ICreateFeature[] createFeatures = featureProvider.getCreateFeatures();
+
+		for (ICreateFeature createFeature : createFeatures) {
+			if (createFeature instanceof FaultTreeNodeCreateFeature) {
+				ObjectCreationToolEntry objectCreationToolEntry = new ObjectCreationToolEntry(
+						createFeature.getCreateName(), createFeature.getCreateDescription(),
+						createFeature.getCreateImageId(), createFeature.getCreateLargeImageId(), createFeature);
+
+				compartmentEntry.addToolEntry(objectCreationToolEntry);
+			}
+		}
+		
+		compartmentEntry = new PaletteCompartmentEntry("Events", null);
+		compartments.add(compartmentEntry);
+		
+		for (ICreateFeature createFeature : createFeatures) {
+			if (!(createFeature instanceof FaultTreeNodeCreateFeature)) {
+				ObjectCreationToolEntry objectCreationToolEntry = new ObjectCreationToolEntry(

```

Comment:
Feels a bit copy&pasty. Maybe a small helper method that takes a createFeature and retuns the ObjectCReationToolEntry?